test/tap/tests/mysql-select_auto_commit-t.cpp

```diff
@@ -0,0 +1,75 @@
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
+#include <unistd.h>
+
+#include <string>
+#include <sstream>
+#include "mysql.h"
+
+#include "tap.h"
+#include "command_line.h"
+#include "utils.h"
+
+int main(int argc, char** argv) {
+	CommandLine cl;
+
+	if (cl.getEnv())
+		return exit_status();
+
+	plan(6);
+
+	MYSQL* mysql = mysql_init(NULL);
+	if (!mysql)
+		return exit_status();
+
+	if (!mysql_real_connect(mysql, cl.host, cl.username, cl.password, NULL, cl.port, NULL, 0)) {
+		fprintf(stderr, "Failed to connect to database: Error: %s\n", mysql_error(mysql));
+		return exit_status();
+	}
+
+	MYSQL_QUERY(mysql, "USE test");
+
+	const char* states[] = {
+		"SELECT @@autocommit",
+		"SET autocommit = 0",
+		"SELECT @@autocommit",
+		"SET autocommit = 1",
+		"SELECT @@autocommit"
+	};
+
+	const char* expected[] = {"1", "0", "1"};
+	int expected_idx = 0;
+
+	for (int i = 0; i < 5; ++i) {
```

Comment:
We can avoid hardcoding the number of queries, but consider the size of `states[]`

Same thing applies for `plan()` : it can be computed based on the content of `states[]` .

The above consideration is related to the fact that:
1. autocommit is changed from 1 (implicit) to 0 and then to 1 (explicitly), but never changed back to 0 (and eventually back to 1) to test multiple changes
2. queries all have `SELECT` uppercase and `autocommit` lowercase : no combination of uppercase and lowercase are tested
3. values `0` and `1` are used , while `ON`/`OFF`/`true`/`FaLsE`/etc are not tested

The test doesn't verify that no backend was used, neither that multiplexing was not disabled.

Reply:
I was going to propose to reuse other test for this purpose. All the feedback from the previous comment can be taken into modifying a test like `reg_test_3549-autocommit_tracking-t.cpp` which performs more complex queries, and already tracks autocommit, so should be a matter of doing an (extra) consistency check between the present tracking and `SELECT @@autocommit`. For addressing the part of the feedback about multiplexing disabling, the utility function `check_multiplexing_disabled` from `test_keep_multiplexing_variables-t.cpp` can also be moved to `utils.cpp` and reused.